dask/dataframe/io/tests/test_parquet.py

```diff
@@ -1699,6 +1699,22 @@ def test_arrow_partitioning(tmpdir):
     ddf.astype({"b": np.float32}).compute()
 
 
+def test_sorted_warnings(tmpdir, engine):
+    tmpdir = str(tmpdir)
+    df = dd.from_pandas(
+        pd.DataFrame({"cola": range(10), "colb": range(10)}), npartitions=2
+    )
+    df.to_parquet(tmpdir, engine=engine, write_index=False)
+    with pytest.warns(RuntimeWarning) as record:
```

Comment:
FYI, you can use `match="'cola', 'colb'"` now I think, rather than the `as record` stuff.

I'm not sure whether it asserts that there is just one assertion. But why do we assert that?

Reply:
There are other warnings too for pyarrow, so I think it needs to be this way

Reply:
I don't think so. I made a small test

```python
import pytest
import warnings


def f(x):
    warnings.warn("test", FutureWarning)
    warnings.warn("user", UserWarning)


def test_foo():
    with pytest.warns(UserWarning, match="user"):
        f(0)
```

and things seem fine with the `match` syntax.

Reply:
What I mean it, I wanted to capture the recorder object, to explicitly count the number of warnings, and make sure that supplying the kwarg makes that number go down, but doesn't go to zero.

Reply:
OK.